### ISSUE TITLE
association_join return wrong :id

### ISSUE THREAD

**Issue by Jesterovskiy** [jeremyevans/sequel#1016]
Example:
``` ruby
class Workout < Sequel::Model
  many_to_one :image
end

class Image < Sequel::Model
  one_to_one :workout
end
```
When I run: `Workout.association_join(:image)[1]` it returns:
```
#<Workout @values={:id=>9, :title=>"*", :description=>"*", :teaser=>"*", :equipment=>"", :video_url=>"*", :video_length=>71446, :met=>nil, :published=>true, :archived=>false, :free=>false, :featured=>false, :trainer_id=>3, :image_id=>9, :level_id=>nil, :created_at=>2015-06-03 16:14:48 +0300, :updated_at=>2015-06-03 16:14:48 +0300, :web_url=>"*", :mobile_url=>"*"}>
```
ID is 9, but I expected to get 1

**Comment by jeremyevans** [comment]
This is expected behavior. The default selection in Sequel is `*`, so you get all columns from all joined tables.  Because Sequel returns results as hashes keyed by column name symbol, if you have multiple columns with the same name in different tables, the later column values will override the early column values if the column name is the same.

As mentioned in the README (http://sequel.jeremyevans.net/rdoc/files/README_rdoc.html#label-Joining+Tables), if you join tables, you should set a specific selection using `Dataset#select` or a similar method.  An alternative is using the `table_select` or `column_select` model plugins, which change the default selection for model datasets.

**Comment by NobodysNightmare** [comment]
I want to raise the question of how that can be _expected_ behaviour once more (FWIW: I did not expect it).

This behaviour easily leads to wrong behaviour by Sequels own helper methods.

I tried today (shortened):

```ruby
Foo.association_left_join(:bar).where(Sequel[:bars][:id] => nil).each do |foo|
  foo.add_bar(...)
end
```

This failed with an error, that the primary key was missing.

At the same time, all places that rely on `foo.id` returning the ID of the `Foo` will build on wrong assumptions, in the worst case using the ID of a `Bar`.

I see that the piece of documentation you are referencing makes sense for "raw SQL building" (i.e. when not using `Sequel` like an ORM), but I feel that when I use Sequel as an ORM, I would expect it to not map columns to the wrong objects in a default use-case. `foo.id` is simply not the ID of `foo`.

**Comment by jeremyevans** [comment]
@NobodysNightmare It's still expected that if you don't set an explicit selection, you get `SELECT *`, and that has issues if you have ambiguous columns in the result set.  If you want different behavior for models, use the `table_select` or `column_select` plugins.

I realize there is potential for the default behavior to cause problems, but changing the default behavior would break working and reasonable code that assumes columns in other tables are available, especially in the case where there is no ambiguity.  It's possible the default behavior could change in Sequel 6, and the current default behavior moved to a plugin.  But there is no chance of that happening other than at a major version bump.

**Comment by NobodysNightmare** [comment]
Yeah that's reasonable. I just wanted to raise my concern that it might not
be expected for everyone.

That said I realize that I found exactly this one issue when looking for
this problem and it is 5 years old with no one else even giving a :+1:

So maybe mine is really an outside perspective. I shared it anyways ;)

Jeremy Evans <notifications@github.com> schrieb am Di., 21. Jan. 2020,
17:38:
